ui/app/components/app/permission-page-container/permission-page-container.component.js

```diff
@@ -0,0 +1,157 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import deepEqual from 'fast-deep-equal'
+import { PermissionPageContainerContent } from '.'
+import { PageContainerFooter } from '../../ui/page-container'
+
+export default class PermissionPageContainer extends Component {
+
+  static propTypes = {
+    approvePermissionsRequest: PropTypes.func.isRequired,
+    rejectPermissionsRequest: PropTypes.func.isRequired,
+    selectedIdentity: PropTypes.object.isRequired,
+    permissionsDescriptions: PropTypes.object.isRequired,
+    domainMetadata: PropTypes.object.isRequired,
+    request: PropTypes.object,
+    redirect: PropTypes.bool,
+    permissionRejected: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    redirect: null,
+    permissionRejected: null,
+    request: {},
+  };
+
+  static contextTypes = {
+    t: PropTypes.func,
+    metricsEvent: PropTypes.func,
+  };
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      selectedPermissions: this.getRequestedMethodState(
+        this.getRequestedMethodNames(props)
+      ),
+      selectedAccount: props.selectedIdentity,
+    }
+  }
+
+  componentDidUpdate () {
+    const newMethodNames = this.getRequestedMethodNames(this.props)
+
+    if (!deepEqual(Object.keys(this.state.selectedPermissions), newMethodNames)) {
+      // this should be a new request, so just overwrite
+      this.setState({
+        selectedPermissions: this.getRequestedMethodState(newMethodNames),
+      })
+    }
+  }
+
+  getRequestedMethodState (methodNames) {
+    return methodNames.reduce(
+      (acc, methodName) => {
+        acc[methodName] = true
+        return acc
+      },
+      {}
+    )
+  }
+
+  getRequestedMethodNames (props) {
+    return Object.keys(props.request.permissions)
+  }
+
+  onPermissionToggle = methodName => () => {
+    this.setState({
+      selectedPermissions: {
+        ...this.state.selectedPermissions,
+        [methodName]: !this.state.selectedPermissions[methodName],
+      },
+    })
+  }
+
+  componentDidMount () {
+    this.context.metricsEvent({
+      eventOpts: {
+        category: 'Auth',
+        action: 'Connect',
+        name: 'Popup Opened',
```

Comment:
- [ ] This metrics event should likely be renamed as this is no longer in a popup